packages/browser/src/lib/on-page-leave.ts

```diff
@@ -0,0 +1,27 @@
+/**
+ * Register event listener on document that fires when:
+ * * tab loses focus or tab close (in mobile or desktop)
+ * * click external link
+ * * click relative link (same domain)
+ * * soft refresh / hard refresh
+ *
+ * adapted from https://stackoverflow.com/questions/3239834/window-onbeforeunload-not-working-on-the-ipad/52864508#52864508,
+ */
+export const onPageLeave = (cb: (...args: any[]) => void) => {
+  let unloaded = false // prevents double firing if both are supported
+
+  // using document instead of window because of bug affecting browsers before safari 14 (detail in footnotes https://caniuse.com/?search=visibilitychange)
+  document.addEventListener('pagehide', () => {
```

Comment:
document instead of window FYI because of an old event bubbling issue (see comment). This _shouldn't_ cause issues that I an forsee, and it's IMO as a rule of thumb it's better to use document rather than window when possible.